core/src/main/java/org/apache/calcite/sql/SqlSelect.java

```diff
@@ -44,6 +44,7 @@
   SqlNode where;
   SqlNodeList groupBy;
   SqlNode having;
+  SqlNode qualify;
```

Comment:
Even though this isn't the pattern used today, let's make this new field private final.